Source/MLX/Ops.swift

```diff
@@ -1741,16 +1909,16 @@ public func tanh(_ array: MLXArray, stream: StreamOrDevice = .default) -> MLXArr
 /// - Parameters:
 ///   - a: input array
 ///   - b: input array
-///   - dimensions: sum over the last `dimensions` dimensions
+///   - axes: sum over the last `axes` dimensions
 ///   - stream: stream or device to evaluate on
 /// - Returns: tensor dot product
 ///
 /// ### See Also
 /// - <doc:arithmetic>
 public func tensordot(
-    _ a: MLXArray, _ b: MLXArray, dimensions: Int = 1, stream: StreamOrDevice = .default
+    _ a: MLXArray, _ b: MLXArray, axes: Int = 1, stream: StreamOrDevice = .default
```

Comment:
Breaking change, but I think important for consistency with the python side.  If needed we could introduce a deprecated cover that uses the old name.